main/editor.js

```diff
@@ -41,7 +42,30 @@ const openEditorWindow = async (filePath, recordFps) => {
 
   loadRoute(editorWindow, 'editor');
 
+  editorWindow.on('close', event => {
+    if (!editorWindow || forceClose) {
+      return;
+    }
+
+    event.preventDefault();
+
+    dialog.showMessageBox(editorWindow, {
+      type: 'question',
+      buttons: ['Discard', 'Cancel'],
+      defaultId: 1,
+      message: 'Are you sure that you want to discard this recording?',
+      detail: 'You will no longer be able to edit and export the original recording.'
+    }, buttonIndex => {
+      if (buttonIndex === 0) {
+        forceClose = true;
```

Comment:
Instead of the `forceClose` thing, you can just unsubscribe to the `close` event here.